File: 605-can-place-flowers/can-place-flowers.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        
        a = [0]

        a.extend(flowerbed)
        a.append(0)

        print(a)
        c = 0
        for i in range(1,len(a)-1):
            if a[i] == 0 and (a[i-1] ==0 and a[i+1] ==0):
                c+=1
                a[i] = 1
        print(c, a)
        if n<=c:
            return True
        return False

File: 605-can-place-flowers/can-place-flowers.java
class Solution {
    public boolean canPlaceFlowers(int[] flowerbed, int n) {
        

        List<Integer> flowers = new ArrayList<>();
        flowers.add(0);
        for(int fl:flowerbed){
            flowers.add(fl);
        }
        flowers.add(0);
        System.out.println(flowers);
        int c=0;
        for(int i=1;i<flowers.size()-1;i++){
            if(flowers.get(i) == 1){
                continue;
            }
            else if(flowers.get(i-1) == 1 || flowers.get(i+1) == 1){
                continue;
            }
            else if(flowers.get(i-1) == 0 && flowers.get(i+1) == 0){
                flowers.set(i,1);
                c+=1;
            }
        }
        System.out.println(c);

        return n<=c;


    }
}